deploy/deploy.config.js

```diff
@@ -0,0 +1,79 @@
+'use strict'
+const path = require('path')
+const fs = require('fs')
+const globalDotEnvFile = path.join(__dirname, '..', '.env')
+
+try {
+  if (fs.existsSync(globalDotEnvFile)) {
+    require('dotenv').config({ path: globalDotEnvFile })
+  } else {
+    console.log('No .env file found at project root')
+  }
+} catch (error) {
+  console.error(error)
+}
+
+const fiveGigabytes = 5120
+// for reference...
+// const fourtyGigabytes = 40960
+// const eightyGigabytes = 81920
+const fourteenDays = 'P14D'
+const fiveMinutes = 'PT5M'
+const oneDay = 'P1D'
+const sixDays = 'P6D'
+
+module.exports = {
+  ServiceBus: {
+    QueueDefaults: {
+      MaxSizeInMegabytes: fiveGigabytes,
+      DefaultMessageTimeToLive: fourteenDays,
+      LockDuration: fiveMinutes,
+      RequiresDuplicateDetection: true,
+      DeadLetteringOnMessageExpiration: true,
+      DuplicateDetectionHistoryTimeWindow: oneDay,
+      EnablePartitioning: false,
+      RequiresSession: false
+    },
+    Queues: [
+      {
+        name: 'check-completion'
+      },
+      {
+        name: 'check-marking'
+      },
+      {
+        name: 'check-notification'
+      },
+      {
+        name: 'check-sync'
+      },
+      {
+        name: 'check-validation'
+      },
+      {
+        name: 'ps-report-schools',
+        DefaultMessageTimeToLive: sixDays,
+        MaxSizeInMegabytes: {}.hasOwnProperty.call(process.env, 'SERVICE_BUS_QUEUE_MAX_SIZE_PS_REPORT') ? parseInt(process.env.SERVICE_BUS_QUEUE_MAX_SIZE_PS_REPORT, 10) : fiveGigabytes
```

Comment:
Consider the failure scenario's here.

By setting the default to the setting fort a lower environment, if we forget to override the setting for prod then the queue will be initialised to 5GB, a figure that is too low.  An error will not be immediately raised and future errors are likely in the overnight runs when there is little oversight.

If we were to make the production values the default, then I believe the deployment would fail immediately and it would be immediately possible to remedy by re-releasing with the override setting.  Releases are watched by several people so it has a high chance of being noticed and corrected.

What do you think?

Reply:
yes, i see the logic here.  Prioritise production over all others.

it does mean that other environments have more default config, but as i have preserved the older scripts it can be staged as necessary.